File: eCinema/eCinema.UI/ecinema_desktop/lib/models/discount.dart
import 'package:json_annotation/json_annotation.dart';

part 'discount.g.dart';

@JsonSerializable()
class Discount {
  final int id;
  final String code;
  final num discountPercentage;
  final DateTime validFrom;
  final DateTime validTo;
  final bool isActive;

  Discount({
    required this.id,
    required this.code,
    required this.discountPercentage,
    required this.validFrom,
    required this.validTo,
    required this.isActive,
  });

  factory Discount.fromJson(Map<String, dynamic> json) =>
      _$DiscountFromJson(json);

  Map<String, dynamic> toJson() => _$DiscountToJson(this);
}

File: eCinema/eCinema.Model/Entities/Discount.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eCinema.Models.Entities
{
    public class Discount
    {
        public int Id { get; set; }
        public string Code { get; set; }
        public decimal DiscountPercentage { get; set; }
        public DateTime ValidFrom { get; set; }
        public DateTime ValidTo { get; set; }
        public bool IsActive { get; set; }

    }
}
